include/dlaf/interface/eigensolver.h

```diff
@@ -0,0 +1,77 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <dlaf/interface/cholesky.h>
+
+#include <pika/init.hpp>
+#include <dlaf/eigensolver/eigensolver.h>
+#include <dlaf/interface/blacs.h>
+#include <dlaf/interface/utils.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_mirror.h>
+
+namespace dlaf::interface {
+
+template <typename T>
+using MatrixMirror = dlaf::matrix::MatrixMirror<T, dlaf::Device::Default, dlaf::Device::CPU>;
```

Comment:
Oh right, we use that in DLA-Future... At least we've commented that HIP doesn't have something similar (https://github.com/eth-cscs/DLA-Future/blob/9fcd6d0bee0817b06557fc9cbc8532966d301994/include/dlaf/eigensolver/band_to_tridiag/mc.h#L395-L398).

I'm of course asking in case we can either:

1. Check that the allocated memory is on the expected device, or
2. Copy the matrix to a GPU if needed.

However, these can be extras. For now we should at least document that we expect the input matrix to be in CPU memory.

Reply:
This has been documented. Do we need to make a note somewhere to re-check this if HIP support becomes available?